src/core_plugins/elasticsearch/lib/ensure_types_exist.js

```diff
@@ -20,19 +20,19 @@ export async function ensureTypesExist({ log, indexName, callCluster, types }) {
   // could be different if aliases were resolved by `indices.get`
   const resolvedName = Object.keys(index)[0];
   const mappings = index[resolvedName].mappings;
-  const literalTypes = Object.keys(mappings);
-  const v6Index = literalTypes.length === 1 && literalTypes[0] === 'doc';
+  const esTypes = Object.keys(mappings);
 
-  // our types aren't really es types, at least not in v6
-  const typesDefined = Object.keys(
-    v6Index
-      ? mappings.doc.properties
-      : mappings
-  );
+  const isV5Index = esTypes.length > 1 || esTypes[0] !== 'doc';
+  if (isV5Index) {
+    throw new Error(
+      'Support for Kibana index format v5 has been removed, reset your ' +
```

Comment:
I like this more as something like 'Your kibana index is out of date, reset it or use the x-pack upgrade assistant."  I mostly don't like the v5 part, because technically it could be v4.